### PR TITLE
feat(azuredevops): implement work items as issues

### LINKED ISSUE #5996: [Feature][Azure devops] Collect Boards from Azure DevOps to supply incidents data for DORA
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Use case

We use Azure repos/boards/pipelines. 

The documentation says:
```
DORA - Median Time to Restore Service
What is this metric?
The time to restore service after service incidents, rollbacks, or any type of production failure happened.
```

and

```
DORA - Change Failure Rate
What is this metric?
The percentage of changes that were made to a code that then resulted in incidents, rollbacks, or any type of production failure.
```

Our 'outage' of services can be seen in our board by viewing tickets with type 'Bug'. There's no other way to report an 'INCIDENT' in Azure devops board. When something is wrong you open a ticket with type='Bug'. You can also set priority (like Critical) and add some labels (like production).

I wish to see DORA metrics in Grafana ; 'Median time to restore service' and 'Change failure rate'. 

### Description

This should be possible by creating a different 'scrapper' for Azure Devops.

### Related issues

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by slaterx**
### Summary
This PR adds support for Azure DevOps work items as issues. The main thing to consider is that in Azure DevOps, work items are connected to projects and orgs, not repositories. Furthermore, current API definitions in Azure DevOps have a few limitations on retrieving work items data. Finally, work item queries are heavy and consume a lot of TSTUs, meaning that a simple query can impact overall organization usage of Azure DevOps and block access to services.

Thus, the decision was to:

- Break down the retrieval into two API calls, one to obtain the work item ids (using the WQL API) and another one to obtain the actual work items in small chunks (using the work item batch API), avoiding throttling issues.
- Use custom code instead of default converter code to store data in issues table.
- Add issue UI components to azure devops page scope configuration.
- Add issue dashboard panels to Azure DevOps grafana page, but remove repository reference from queries.

### Does this close any open issues?
Closes #5996

### Screenshots
image

### Other Information
With all work items collected, and their boards stored as component, having incidents mapped is as easy as querying work items with `type = bug`.

### CODE REVIEW

File: backend/plugins/azuredevops_go/tasks/workitem_collector.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	logger.Debug("All Ids received: #%s", rawWorkItems.String())
+
+	logger.Info("Sum of workitems Ids received: %d", len(rawWorkItems.Array()))
+
+	var currentWorkItems, emptyWorkItems []string
+
+	rawWorkItems.ForEach(func(key, value gjson.Result) bool {
+		currentWorkItems = append(currentWorkItems, value.String())
+
+		if len(currentWorkItems) == 9 || key.Int() == int64(len(rawWorkItems.Array())-1) {
```
Comment (mr-ks) [inline backend/plugins/azuredevops_go/tasks/workitem_collector.go:86]: Can you please explain how you came up with this condition? Is this a restriction of the `/workitemsbatch` API?

**Review by mr-ks — CHANGES_REQUESTED** [review]
Could you please check the GitHub Actions workflow? Some of the checks are failing e.g the [golangci-lint check](https://github.com/apache/incubator-devlake/actions/runs/10190826555)

**Comment by magiskboy** [comment]
have any progess in this pr?

**Comment by klesh** [comment]
> have any progess in this pr?

It was closed in 2024.

**Pull request closed without merging**